mmv1/templates/terraform/post_create/datastream_stream.go.erb

```diff
@@ -14,4 +14,11 @@
 -%>
 if err := waitForDatastreamStreamReady(d, config, d.Timeout(schema.TimeoutCreate) - time.Minute); err != nil {
   return fmt.Errorf("Error waiting for Stream %q to be NOT_STARTED or RUNNING during creation: %q", d.Get("name").(string), err)
-}
\ No newline at end of file
+}
+
+if d.Get("state") != d.Get("desired_state") {
+  log.Printf("[DEBUG] Desired state %s not equal to state = %s, updating stream %q", d.Get("desired_state"), d.Get("state"), d.Id())
+  if err = resourceDatastreamStreamUpdate(d, meta); err != nil {
```

Comment:
I'm not seeing how calling the `resourceDatastreamStreamUpdate` will affect the `state`, other than adding in another wait. `resourceDatastreamStreamUpdate` will check if any fields need to be updated, but `desired_state` will still be the same as what it is here. 
What will the `resourceDatastreamStreamUpdate` actually do? How have you tested this scenario?

Reply:
Yes, this is tested.

Datastream actually does not support create Stream with state=RUNNING and will always return state=NOT_STARTED

So we need to run another update call so that state will be: RUNNING (only update calls can set sate=RUNNING)